tests/unittest_builder.py

```diff
@@ -949,7 +956,60 @@ def test_arguments_of_signature() -> None:
     """Test that arguments is None for function without an inferable signature."""
     node = builder.extract_node("int")
     classdef: nodes.ClassDef = next(node.infer())
-    assert all(i.args.args is None for i in classdef.getattr("__dir__"))
+    assert all(i.args.args == [] for i in classdef.getattr("__dir__"))
+    assert all(len(i.args.posonlyargs) == 1 for i in classdef.getattr("__dir__"))
+
+
+def test_c_module_text_signature() -> None:
+    """Test for c-extensions with invalid __text_signature__ attribute."""
+
+    def _find_dot_so_files(path: Path) -> Iterator[Path]:
+        for entry in path.iterdir():
+            if entry.suffix in importlib.machinery.EXTENSION_SUFFIXES:
+                yield entry
+
+    def _import_module(path: Path, module_full_name: str) -> types.ModuleType:
+        spec = importlib.util.spec_from_file_location(module_full_name, path)
+        if spec is None:
+            raise RuntimeError(
+                f"Cannot find spec for module {module_full_name} at {path}"
+            )
+        py_mod = importlib.util.module_from_spec(spec)
+        loader = spec.loader
+        assert isinstance(loader, importlib.abc.Loader), loader
+        loader.exec_module(py_mod)
+        return py_mod
+
+    c_module_invalid_text_signature = (
+        Path(resources.RESOURCE_PATH) / "c_module_invalid_text_signature"
+    )
+    package_path = c_module_invalid_text_signature / "mymod"
+
+    # build extension
+    try:
+        cwd = os.getcwd()
+        code, outstr = subprocess.getstatusoutput(
+            f"cd {c_module_invalid_text_signature} && python3 setup.py build_ext --inplace"
+        )
+        os.chdir(cwd)
```

Comment:
I don't like that this is done in the repo file tree. Cleanup will always be difficult. Even the `cwd` reset here isn't safe if `subprocess.getstatusoutput` throws an exception.

Maybe take a look at the `tempdir` contextmanager from `pylint -> lint/unittest_lint.py` and also [tempfile.TemporaryDirectory](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory).

Reply:
There's [a pytest fixture named tmpdir](https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmpdir-fixture) that would be useful here. What is done in pylint is a little dated but refactoring everything is not a top priority.